app/jobs/cache_analytics_job.rb

```diff
@@ -5,38 +5,27 @@ class CacheAnalyticsJob < ApplicationJob
   queue_as :default
 
   def perform(*_args)
-    Analytic.transaction do
-      analytics_by_jur_path = {}
-      root_jurs = []
+    Jurisdiction.find_each do |jur|
+      Analytic.transaction do
+        analytic = Analytic.new(jurisdiction_id: jur.id)
 
-      leaf_nodes = Jurisdiction.all.select { |jur| jur.has_children? == false }
-      leaf_nodes.each do |leaf_jurisdiction|
-        leaf_analytic = self.class.calculate_analytic_local_to_jurisdiction(leaf_jurisdiction)
-        analytics_by_jur_path[leaf_jurisdiction[:path]] = leaf_analytic
-        # Start recursive bubble up of analytic data
-        self.class.add_analytic_to_parent(leaf_jurisdiction, leaf_analytic, analytics_by_jur_path)
-      end
-
-      # Map data will be on the top-level jurisdiction only
-      root_nodes = Jurisdiction.where(ancestry: nil)
-      root_nodes.each do |root_jurisdiction|
-        symp_by_state = root_jurisdiction.all_patients.pluck(:address_state).each_with_object(Hash.new(0)) { |state, counts| counts[state] += 1 }
-        monitored_by_state = root_jurisdiction.all_patients.symptomatic.uniq.pluck(:address_state).each_with_object(Hash.new(0)) do |state, counts|
-          counts[state] += 1
+        # Map data will be on the top-level jurisdiction only (will be removed once new maps are implemented)
+        if jur.root?
+          symp_by_state = jur.all_patients.pluck(:address_state).each_with_object(Hash.new(0)) { |state, counts| counts[state] += 1 }
+          monitored_by_state = jur.all_patients.symptomatic.uniq.pluck(:address_state).each_with_object(Hash.new(0)) do |state, counts|
```

Comment:
In this call the `.uniq` call is redundant as `jur.all_patients.symptomatic` produces a `SELECT DISTINCT` query.
```sql
SELECT DISTINCT `patients`.* FROM `patients` LEFT OUTER JOIN `assessments` ON `assessments`.`patient_id` = `patients`.`id` WHERE `patients`.`jurisdiction_id` IN (1, 2, 5, 3, 4, 6, 7) AND `patients`.`monitoring` = TRUE AND `patients`.`purged` = FALSE AND `patients`.`public_health_action` = 'None' AND (assessments.symptomatic = TRUE)
```

```ruby
a = j.all_patients.symptomatic.uniq
b = j.all_patients.symptomatic
a == b # => true
```

Reply:
yeah, so this was the old code that I just moved to another block for now (which will also be removed as soon as David's front end analytics stuff is ready to go), but I will still put this change in. thanks!